### PR TITLE
Refactor CLI commands

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
Refactor CLI commands

### CODE REVIEW

File: components/cli/cmd/cellery/logs.go [hunk 1]
```diff
@@ -39,21 +41,21 @@ func newLogsCommand() *cobra.Command {
...
 					return fmt.Errorf("expects a valid component name, received %s", args[0])
 				}
 			}
 			return nil
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := commands.RunLogs(args[0], component, syslog); err != nil {
-				return fmt.Errorf("error running cellery log, %v", err)
+			if err := instance.RunLogs(cli, args[0], component, syslog); err != nil {
+				util.ExitWithErrorMessage("Cellery logs command failed", err)
```
Comment (isurulucky) [inline components/cli/cmd/cellery/logs.go]: Why are we exiting with error if we are using RunE here?

File: components/cli/internal/test/bal_executor.go [hunk 2]
```diff
@@ -47,15 +52,54 @@ func SetBalVersion(version string) func(*MockBalExecutor) {
...
+
+func SetYamlName(name string) func(*MockBalExecutor) {
+	return func(balExecutor *MockBalExecutor) {
+		balExecutor.yamlName = name
+	}
+}
+
+func SetYamlContent(content []byte) func(*MockBalExecutor) {
+	return func(balExecutor *MockBalExecutor) {
+		balExecutor.yamlContent = content
```
Comment (isurulucky) [inline components/cli/internal/test/bal_executor.go]: Why do we need yaml content for ballerina executor?

File: components/cli/internal/test/bal_executor.go [hunk 3]
```diff
@@ -47,15 +52,54 @@ func SetBalVersion(version string) func(*MockBalExecutor) {
...
+	}
+}
+
+func SetMetadataJsonContent(content []byte) func(*MockBalExecutor) {
+	return func(balExecutor *MockBalExecutor) {
+		balExecutor.metadataJsonContent = content
```
Comment (isurulucky) [inline components/cli/internal/test/bal_executor.go]: Why do we need json content in bal executor?

File: components/cli/pkg/commands/image/build.go [hunk 4]
```diff
@@ -71,33 +70,33 @@ func RunBuild(cli cli.Cli, tag string, fileName string) error {
...
 			err := generateMetaData(cli, parsedCellImage, projectDir)
 			return err
 		}); err != nil {
 		return err
 	}
-	// Create the artifacts zip file
+	// Create the artifacts zip file.
 	artifactsZip := parsedCellImage.ImageName + cellImageExt
-	if err = cli.ExecuteTask("Creating cell image zip file", "Failed to create temporary baf file",
+	if err = cli.ExecuteTask("Creating cell image zip file", "Failed to image zip",
```
Comment (isurulucky) [inline components/cli/pkg/commands/image/build.go]: s/Failed to image zip/failed to create image zip/

File: components/cli/cmd/cellery/logs.go (hunk 1 above)
Comment (MadhukaHarith92) [reply components/cli/cmd/cellery/logs.go]: Updated. Not using RunE, instead using util.ExitWithErrorMessage.

File: components/cli/pkg/commands/image/build.go (hunk 4 above)
Comment (MadhukaHarith92) [reply components/cli/pkg/commands/image/build.go]: Updated the error message to "Failed to create the image zip"

File: components/cli/internal/test/bal_executor.go (hunk 2 above)
Comment (MadhukaHarith92) [reply components/cli/internal/test/bal_executor.go]: This is the mock ballerina executor. For unit tests, we need to mock the generation of artifacts files (metadata.json & yaml) to test "Cellery Run" command.

File: components/cli/internal/test/bal_executor.go (hunk 3 above)
Comment (MadhukaHarith92) [reply components/cli/internal/test/bal_executor.go]: This is the mock ballerina executor. For unit tests, we need to mock the generation of artifacts files (metadata.json & yaml) to test "Cellery Run" command.

File: components/cli/internal/test/bal_executor.go (hunk 2 above)
Comment (isurulucky) [reply components/cli/internal/test/bal_executor.go]: Understood, thanks.

File: components/cli/internal/test/bal_executor.go (hunk 3 above)
Comment (isurulucky) [reply components/cli/internal/test/bal_executor.go]: Ack, thanks.

**Pull request merged**